examples/udp_server.rs

```diff
@@ -1,58 +1,75 @@
+        // Poll to check if we have events waiting for us.
         poll.poll(&mut events, None)?;
-        // If we do iterate throuigh them
+
+        // Process each event.
         for event in events.iter() {
             // Validate the token we registered our socket with,
             // in this example it will only ever be one but we
             // make sure it's valid none the less.
             match event.token() {
-                IN => loop {
-                    // In this loop we receive from the socket as long as we
-                    // can read data
+                UDP_SOCKET => loop {
+                    // In this loop we receive all packets queued for the socket.
                     match socket.recv_from(&mut buf) {
-                        Ok((n, from_addr)) => {
-                            // Send the data right back from where it came from.
-                            socket.send_to(&buf[..n], from_addr)?;
+                        Ok((packet_size, source_address)) => {
+                            // Echo the data.
+                            socket.send_to(&buf[..packet_size], source_address)?;
+                        }
+                        Err(e) if e.kind() == io::ErrorKind::WouldBlock => {
+                            // If we get a `WouldBlock` error we know our socket
+                            // has no more packets queued, so we can return to
+                            // polling and wait for some more.
+                            break;
                         }
                         Err(e) => {
-                            // If we failed to receive data we have two cases
-                            if e.kind() == io::ErrorKind::WouldBlock {
-                                // If the reason was `WouldBlock` we know
-                                // our socket has no more data to give so
-                                // we can return to the poll to wait politely.
-                                break;
-                            } else {
-                                // If it was any other kind of error, something
-                                // went wrong and we terminate with an error.
-                                return Err(e);
-                            }
+                            // If it was any other kind of error, something went
+                            // wrong and we terminate with an error.
+                            return Err(e);
                         }
                     }
                 },
-                // We only have IN as a token, so this should never ever be hit
-                _ => unreachable!(),
+                _ => {
+                    // This should never happen as we only registered our
+                    // `UdpSocket` using the `UDP_SOCKET` token, but if it ever
+                    // does we'll log it.
+                    warn!("Got event for unexpected token: {:?}", event);
```

Comment:
shouldn't it throw error?

Reply:
I wouldn't advice to so. Although it should never happen, you probably don't want your production application to go down just because it received an event it can easily ignore.

Reply:
Hmm it should be error! instead of warn!?

Reply:
That depends on your philosophy on logging I guess. Personally I use `error!` for things that need intimidate attention, for me this doesn't fall in that category.